private/pkg/oauth2/client.go

```diff
@@ -0,0 +1,181 @@
+// Copyright 2020-2024 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package oauth2
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"mime"
+	"net/http"
+	"strings"
+	"time"
+)
+
+// Client is an OAuth 2.0 client that can register a device, authorize a device,
+// and poll for the device access token.
+type Client struct {
+	baseURL string
+	client  *http.Client
+}
+
+// NewClient returns a new Client with the given base URL and HTTP client.
+func NewClient(baseURL string, client *http.Client) *Client {
+	return &Client{
+		baseURL: baseURL,
```

Comment:
Nit, could strip a potential `/` from the suffix to avoid `c.baseURL+DeviceRegistrationPath` failing later on.